### PR TITLE
Fix printing guidelines

### PR DESCRIPTION
**Pull request by fingolfin**
Properly explain how to reach each printing mode, and also explain what 'io' is

### CODE REVIEW

File: docs/src/DeveloperDocumentation/printing_details.md [hunk 1]
```diff
@@ -7,15 +7,17 @@ found in the [Developer Style Guide](@ref).
 
 ## Implementing show functions
 
-Here is the translation between `:detail`, `one line` and `:supercompact`.
+Here is the translation between `:detail`, `one line` and `:supercompact`,
+where `io` is an `IO` object (such as `stdout` or an `IOContext`):
```
Comment (fingolfin) [inline docs/src/DeveloperDocumentation/printing_details.md]:
```suggestion
where `io` is an `IO` object (such as `stdout` or an `IOBuffer`):
```

File: docs/src/DeveloperDocumentation/printing_details.md [hunk 2]
```diff
@@ -173,7 +175,7 @@ end
 
 function Base.show(io::IO, b::B)
   io = AbstractAlgebra.pretty(io)
-  print(io, LowercaseOff(), "Hilbert thing")
+  print(io, AbstractAlgebra.LowercaseOff(), "Hilbert thing")
```
Comment (simonbrandhorst) [inline docs/src/DeveloperDocumentation/printing_details.md:178]: I believe this is exported? Then `AbstractAlgebra.` is not necessary.

Comment (ThomasBreuer) [reply docs/src/DeveloperDocumentation/printing_details.md:178]: As far as I understand, `LowercaseOff`, `Indent`, `Dedent`, `pretty` are imported from AbstractAlgebra into Oscar, thus one need not write `AbstractAlgebra.` in code inside Oscar; but they are not exported from Oscar.

**Comment by thofma** [comment]
Looks good, once the the namespace has been figured out.

File: docs/src/DeveloperDocumentation/printing_details.md (hunk 2 above)
Comment (fingolfin) [reply docs/src/DeveloperDocumentation/printing_details.md:178]: The example code uses an explicit `AbstractAlgebra.` everywhere else, and it should be consistent. With it there, a user can copy&paste the example code and start working with it, whether or not they have done `using AbstractAlgebra` (which is dangerous inside a package) or `import AbstractAlgebra`.

Anyway, my primary concern was to make it consistent. If someone prefers to drop the `AbstractAlgebra.` everywhere, that should be a seperate PR, IMHO

**Review by fieker — APPROVED** [review]

**Pull request merged**